tests/view_tests/tests/test_debug.py

```diff
@@ -375,6 +375,21 @@ def test_reporting_of_nested_exceptions(self):
         self.assertIn(implicit_exc.format("Second exception"), text)
         self.assertEqual(3, text.count('<p>Final exception</p>'))
 
+    def test_reporting_frames_without_source(self):
+        try:
+            source = "def funcName():\n    raise Error('Whoops')\nfuncName()"
```

Comment:
(This particular string is fine but use single quotes in "funcName" later in the test.)